jkube-kit/resource/helm/src/main/java/org/eclipse/jkube/kit/resource/helm/HelmRepositoryUploader.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.resource.helm;
+
+import io.fabric8.kubernetes.client.RequestConfigBuilder;
+import io.fabric8.kubernetes.client.http.HttpClient;
+import io.fabric8.kubernetes.client.http.HttpRequest;
+import io.fabric8.kubernetes.client.http.HttpResponse;
+import io.fabric8.kubernetes.client.utils.HttpClientUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jkube.kit.common.KitLogger;
+import org.eclipse.jkube.kit.common.util.Base64Util;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static org.eclipse.jkube.kit.common.util.AsyncUtil.getCompletableFuture;
+
+public abstract class HelmRepositoryUploader implements HelmUploader {
+  public abstract String method();
+
+  public abstract KitLogger getLogger();
+
+  public abstract String url(File helmChart, HelmRepository repository);
+
+  @Override
+  public void uploadSingle(File file, HelmRepository repository) throws IOException, BadUploadException {
+    String uploadUrl = url(file, repository);
+
+    try (HttpClient httpClient = HttpClientUtils.getHttpClientFactory().newBuilder().tag(new RequestConfigBuilder().withRequestRetryBackoffLimit(0).build()).build()) {
+      HttpRequest.Builder httpRequestBuilder = httpClient.newHttpRequestBuilder();
+      httpRequestBuilder.uri(uploadUrl);
+      httpRequestBuilder.header("Authorization", String.format("Basic %s", Base64Util.encodeToString(repository.getUsername() + ":" + repository.getPassword())));
```

Comment:
Add a comment here to remind us that at this point username and password are always populated since this is a requirement from our HelmService